### PR TITLE
bugfix in population worker and key_source of ComputeTraces

### PR DESCRIPTION
**Pull request by fabiansinz**
- When the key_source of a computed relation was not properly defined, populate would do nothing but progress would still show that the relation is not fully populated. Because of that, the worker constantly tried to populate the remaining items, jamming the server. I added a time condition to account for that. 

- `preprocess.ComputeTraces`  had exactly that problem. Sometimes there were just no traces for a particular scan. I fixed that in the `key_source`.

### CODE REVIEW

File: python/scripts/worker-populate.py [hunk 1]
```diff
@@ -41,13 +41,16 @@ def main(argv):
     run_daemon = True # execute at least one loop
     while run_daemon:
         populated_at_least_one = False
+        start = time.time()
         for name, rel in rels_cls.items():
             todo, done = rel().progress()
             if todo > 0:
                 populated_at_least_one = True
                 rel().populate(reserve_jobs=True)
-        if run_daemon and not populated_at_least_one:
```
Comment (dimitri-yatsenko) [inline python/scripts/worker-populate.py]: Why not always wait?  If it attempted to populate, then there is probably no need to check again immediately, right?

Comment (fabiansinz) [inline python/scripts/worker-populate.py]: Hmm, I guess you're right. I'll change it later.

Comment (dimitri-yatsenko) [inline python/scripts/worker-populate.py]: thanks. it should simplify the logic.

File: python/scripts/worker-populate.py [hunk 2]
```diff
@@ -41,13 +41,16 @@ def main(argv):
     run_daemon = True # execute at least one loop
     while run_daemon:
         populated_at_least_one = False
+        start = time.time()
         for name, rel in rels_cls.items():
             todo, done = rel().progress()
```
Comment (dimitri-yatsenko) [inline python/scripts/worker-populate.py]: progress can be a time consuming operator and its function is duplicated in populate.  I would just run populate without checking progress.

File: python/scripts/worker-populate.py [hunk 3]
```diff
@@ -41,13 +41,16 @@ def main(argv):
...
             if todo > 0:
                 populated_at_least_one = True
                 rel().populate(reserve_jobs=True)
```
Comment (dimitri-yatsenko) [inline python/scripts/worker-populate.py]: need to have the option to suppress errors.  populate does not catch errors by default.

**Pull request merged**